src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -49,7 +49,22 @@ export class UserService {
     private readonly geoLocationService: GeoLocationService,
     private readonly countryService: CountryService,
     private readonly cryptoService: CryptoService,
-  ) {}
+  ) {
+    this.setAllUserBlockchain();
+  }
+
+  async setAllUserBlockchain(): Promise<void> {
+    const users = await this.userRepo.find({ where: { blockchain: IsNull() } });
+    for (const user of users) {
+      user.blockchain = this.cryptoService.getBlockchainsBasedOn(user.address);
+
+      try {
+        await this.userRepo.update(user.id, { blockchain: user.blockchain });
+      } catch (e) {
+        console.error(`Error updating initial user blockchain for user id: ${user.id}`, e);
+      }
+    }
+  }
```

Comment:
This does not really work, since it would assign all BSC users to Ethereum. Maybe we can find the right blockchain looking at the buy routes of the user?